backend/composer/services/cs_ingestion/cs_ingestion_services.py

```diff
@@ -1,292 +1,523 @@
+            if getattr(connectivity_statement, field) != value:
+                return True
+
+    # Check for changes in species
+    current_species = set(species.ontology_uri for species in connectivity_statement.species.all())
+    new_species = set(statement.get(SPECIES, []))
+    if current_species != new_species:
+        return True
 
+    # Check for changes in provenance
+    current_provenance = set(provenance.uri for provenance in connectivity_statement.provenance_set.all())
+    new_provenance = set(statement.get(PROVENANCE, []))
+    if current_provenance != new_provenance:
+        return True
 
-def do_transition_to_compose_now(sentence: Sentence, user: User):
-    available_transitions = [
-        available_state.target
-        for available_state in sentence.get_available_user_state_transitions(
-            user
-        )
+    # Check for changes in forward_connection
+    current_forward_connections = set(connection.id for connection in connectivity_statement.forward_connection.all())
+    new_forward_connections = set(statement.get(FORWARD_CONNECTION, []))
+    if current_forward_connections != new_forward_connections:
+        return True
+
+    # Check for changes in origins
+    current_origins = set(origin.ontology_uri for origin in connectivity_statement.origins.all())
+    new_origins = statement[ORIGINS].anatomical_entities
+    if current_origins != new_origins:
+        return True
+
+    # Check for changes in vias
+    current_vias = [
+        {
+            'anatomical_entities': set(via.anatomical_entities.all().values_list('ontology_uri', flat=True)),
+            'from_entities': set(via.from_entities.all().values_list('ontology_uri', flat=True))
+        }
+        for via in connectivity_statement.via_set.order_by('order').all()
     ]
-    if SentenceState.COMPOSE_NOW in available_transitions:
-        # we need to update the state to compose_now when the system user has the permission to do so
-        sentence = SentenceService(sentence).do_transition(
-            SentenceState.COMPOSE_NOW, user
-        )
-        sentence.save()
-    return sentence
+    new_vias = statement[VIAS]
 
+    if len(current_vias) != len(new_vias):
+        return True
 
-def do_transition_to_exported(statement: ConnectivityStatement, user: User):
-    cs = ConnectivityStatementService(statement).do_transition(
-        CSState.EXPORTED, user
+    for current_via, new_via in zip(current_vias, new_vias):
+        if (new_via.anatomical_entities != current_via['anatomical_entities'] or
+                new_via.from_entities != current_via['from_entities']):
+            return True
+
+    # Check for changes in destinations
+    current_destinations = connectivity_statement.destinations.all()
+    new_destinations = statement[DESTINATIONS]
+
+    if len(current_destinations) != len(new_destinations):
+        return True
+
+    # We may need to change this algorithm when multi-destination is supported by neurondm
```

Comment:
The issue here is that multi-destinations have no way to identify themselves, opposed to vias that have the order.
Currently I check if the set of all destinations from neurondm matches with the set of all destinations from composer (and because there's only 1 in neurondm, it will work), but if at some point there's more than one, we may be interested in a different approach. cc @ddelpiano